fastparquet/writer.py

```diff
@@ -571,6 +574,12 @@ def write(filename, data, partitions=[0], encoding="PLAIN",
                     fmd.row_groups.append(rg)
             fmd.num_rows = sum(rg.num_rows for rg in fmd.row_groups)
         else:
+            if write_index or write_index is None and not (
```

Comment:
Ideally, this should live lower, in make_row_group, which gets called from here and from the groupby thing above (which I see looks a bit ugly) and by the dask writer too. Otherwise, it will have to be repeated.